app/models/user.rb

```diff
@@ -27,29 +27,51 @@ class User < ActiveRecord::Base
 
   has_one :member
 
-  def self.from_omniauth(auth)
-    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
-      user.email = auth.info.email
-      user.password = Devise.friendly_token[0, 20]
+  class << self
+    def from_omniauth(auth)
+      find_with_auth_provider(auth) ||
+        find_existing_with_auth_email(auth) ||
+        create_with_auth(auth)
     end
-  end
 
-  def self.new_with_session(params, session)
-    super.tap do |user|
-      if data = session["devise.github_data"] &&
-                session["devise.github_data"]["extra"]["raw_info"]
-        user.email = data["email"] if user.email.blank?
+    def new_with_session(params, session)
+      super.tap do |user|
+        if data = session["devise.github_data"] &&
+                  session["devise.github_data"]["extra"]["raw_info"]
+          user.email = data["email"] if user.email.blank?
+        end
       end
     end
-  end
 
-  def self.create_default_user!
-    User.create! email: User::DEFAULT_USER_EMAIL, password: "12345678"
-  end
+    def create_default_user!
+      User.create! email: User::DEFAULT_USER_EMAIL, password: "12345678"
+    end
 
-  def self.default_user
-    User.find_by(email: DEFAULT_USER_EMAIL)
-  end
+    def default_user
+      User.find_by(email: DEFAULT_USER_EMAIL)
+    end
+
+    private
```

Comment:
They actually are private class methods because they are inside of `class << self`